synapse/util/frozenutils.py

```diff
@@ -35,7 +36,7 @@ def freeze(o: Any) -> Any:
 
 
 def unfreeze(o: Any) -> Any:
```

Comment:
I note that:
- this doesn't unfreeze a frozenset to a set
- maybe the try-except could be replaced with an `if isinstance(o, collections.abc.Iterable)`?
- unfreezing a string bytestring still leaves you with an immutable object!

Judging from where we use it, it looks like this is only used to unfreeze event content rather than a generic unfreeze function.

Reply:
I don't believe it is meant to be a generic `unfreeze` function. 👍